CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
Lets for example, a custom field has value  
 ``` 'CRM_Core_DAO::VALUE_SEPARATOR�Colorado Mountain Bike Association (COMBA)�CRM_Core_DAO::VALUE_SEPARATOR'. ```
1. So yes multi-select values are always separated by VALUE_SEPARATOR and it is itself a control character. So using [[:cntrl:]] we can handle the presence of both VALUE_SEPARATOR and any Control/special character in value (like in above �). 
    If some how one uses exact value to retrieve via api/Search Builder then [[:cntrl:]]* is flexible enough to handle 0 or multiple occurrences of cntrl characters in value. 
2, Yes unfortunately the existing regex ONLY match cntrl/special characters at start and end of the value.